aether-kernel/aether/kernel/api/models.py

```diff
@@ -246,16 +246,15 @@ class Submission(ExportModelOperationsMixin('kernel_submission'), ProjectChildAb
 
     mappingset = models.ForeignKey(
         to=MappingSet,
-        on_delete=models.CASCADE,
+        on_delete=models.SET_NULL,
```

Comment:
Linked to this change and to ensure that the multitenancy will working you need to remove this method in this model

```python
    def get_mt_instance(self):
        # because project can be null we need to override the method
        return self.mappingset.project
```

You need to adapt also the serializer and the ViewSet.

Reply:
Another required change...
```python
def save(self, *args, **kwargs):
        self.project = self.mappingset.project  # <-- this fails if mappingset was deleted!!!
        super(Submission, self).save(*args, **kwargs)
```